ext/intl/rangeformatter/rangeformatter_class.cpp

```diff
@@ -0,0 +1,221 @@
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Bogdan Ungureanu <bogdanungureanu21@gmail.com>              |
+   +----------------------------------------------------------------------+
+*/
+
+#include <unicode/numberrangeformatter.h>
+#include <unicode/unumberrangeformatter.h>
+#include <unicode/numberformatter.h>
+#include <unicode/unistr.h>
+#include "../intl_convertcpp.h"
+
+extern "C" {
+    #include "php.h"
+    #include "zend_API.h"
+    #include "../intl_common.h"
+    #include "../intl_error.h"
+    #include "../php_intl.h"
+    #include "../intl_data.h"
+    #include "rangeformatter_arginfo.h"
+    #include "rangeformatter_class.h"
+#include "intl_convert.h"
+}
+
+using icu::number::NumberRangeFormatter;
+using icu::number::NumberFormatter;
+using icu::number::UnlocalizedNumberFormatter;
+using icu::number::LocalizedNumberRangeFormatter;
+using icu::UnicodeString;
+using icu::MeasureUnit;
+
+static zend_object_handlers rangeformatter_handlers;
+zend_class_entry *class_entry_IntlNumberRangeFormatter;
+
+zend_object *IntlNumberRangeFormatter_object_create(zend_class_entry *ce)
+{
+	IntlNumberRangeFormatter_object* intern;
+
+	intern = (IntlNumberRangeFormatter_object*)zend_object_alloc(sizeof(IntlNumberRangeFormatter_object), ce);
+	zend_object_std_init(&intern->zo, ce);
+	object_properties_init(&intern->zo, ce);
+
+	// Initialize rangeformatter_data structure
+	intl_error_init(&intern->nrf_data.error);
+	intern->nrf_data.unumrf = nullptr;
+
+    intern->zo.handlers = &rangeformatter_handlers;
+
+	return &intern->zo;
+}
+
+U_CFUNC PHP_METHOD(IntlNumberRangeFormatter, __construct)
+{
+    ZEND_PARSE_PARAMETERS_NONE();
+    zend_throw_error(NULL, "Cannot directly construct %s, use createFromSkeleton method instead", ZSTR_VAL(Z_OBJCE_P(ZEND_THIS)->name));
+}
+
+U_CFUNC PHP_METHOD(IntlNumberRangeFormatter, createFromSkeleton)
+{
+    #if U_ICU_VERSION_MAJOR_NUM < 63
+        zend_throw_error(NULL, "IntlNumberRangeFormatter is not available in ICU 62 and earlier");
+        RETURN_THROWS();
+    #endif
+
+    char* skeleton;
+    char* locale;
+    size_t locale_len;
+    size_t skeleton_len;
+    zend_long collapse;
+    zend_long identityFallback;
+
+    ZEND_PARSE_PARAMETERS_START(4,4)
+        Z_PARAM_STRING(skeleton, skeleton_len)
+        Z_PARAM_STRING(locale, locale_len)
+        Z_PARAM_LONG(collapse)
+        Z_PARAM_LONG(identityFallback)
+    ZEND_PARSE_PARAMETERS_END();
+
+    if (locale_len == 0) {
+        locale = (char *)intl_locale_get_default();
+    }
+
+    if (skeleton_len == 0) {
```

Comment:
nit: preferably swap this condition with the one above so you do not bother worrying with locale if you do not have the required skeleton to begin with ;)